serverFiles/models/articles.models.js

```diff
@@ -18,14 +18,43 @@ function selectArticlesById(article_id) {
     });
 }
 
-function selectAllArticles() {
+function selectAllArticles(filterBy, sortBy, orderBy) {
+  const validSortBy = ['comment_count', 'author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url']
+    if(!validSortBy.includes(sortBy)) {
+      return Promise.reject({status:400,msg:'Bad request'})
+    }
+
+  const validFilterBy = ['mitch', 'cats', 'paper']
+    if (filterBy) {
+      if (!validFilterBy.includes(filterBy)) {
+        return Promise.reject({status:400,msg:'Bad request'})
+      }
+    }
+
+    const validOrderBy = ['asc', 'desc'];
+    if (orderBy) {
+      if (!validOrderBy.includes(orderBy)) {
+        return Promise.reject({status:400, msg:'Bad request'})
+      }
+    }
+
+  const queryValues = []
+
+  let queryString = `SELECT COUNT(comments.body) AS comment_count, articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url  FROM articles JOIN comments ON articles.article_id = comments.article_id `
+  
+  if (filterBy) {
```

Comment:
This is a nice way of handling whether or not there is a filter by, love the use of the $ array here.